source/lambda/pyproject.toml

```diff
@@ -0,0 +1,42 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+[build-system]
+requires = ["setuptools"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.packages.find]
+where = [""]
+exclude = [
+     "tests"
+]
+
+[project]
+name = "InstanceScheduler-Lambda"
+version = "1.5"
+description = "AWS Instance Scheduler"
+
+dependencies = [
+    "pytz",
+    "requests",
+    "boto3",
+    "aws-cdk-lib==2.60.0",
+    "constructs>=10.0.0,<11.0.0",
+]
+
+[project.optional-dependencies]
+dev = [
+    "tox",
+    "boto3-stubs-lite[essential]",
+    "boto3-stubs-lite[ssm]",
+    "botocore>=1.12.99",
+    "boto3>=1.9.99",
```

Comment:
Unless you package the sdk in your lambda or layer yourself, you get a specific version: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

I don't see a reason to test locally with a different version.

Reply:
you're probably right. the dependencies list here definitely needs to be pruned and consolidated (it's currently split between the pyproject.toml and requirements/testing_requirements .txt files) Cleaning up the dependencies into a single source of truth would be the scope of the upcoming updates to the build process (my goal here was just to tack a working test process onto the existing build process)